src/v/redpanda/admin/security.cc

```diff
@@ -813,3 +812,32 @@ admin_server::update_role_handler(std::unique_ptr<ss::http::request> req) {
     j_res.role = to_role_name();
     co_return ss::json::json_return_type(j_res);
 }
+
+ss::future<ss::json::json_return_type>
+admin_server::delete_role_handler(std::unique_ptr<ss::http::request> req) {
+    if (need_redirect_to_leader(model::controller_ntp, _metadata_cache)) {
+        // In order that we can do a reliably ordered validation of
+        // the request (and drop no-op requests), run on controller leader;
+        throw co_await redirect_to_leader(*req, model::controller_ntp);
+    }
+
+    ss::sstring role_v;
+    if (!admin::path_decode(req->param["role"], role_v)) {
+        vlog(adminlog.debug, "Invalid parameter 'role' got {}", role_v);
+        throw ss::httpd::bad_param_exception{fmt::format(
+          "Invalid parameter 'role' got {{{}}}", req->param["role"])};
+    }
+    auto role_name = security::role_name(role_v);
```

Comment:
Is it worth extracting this as a function and reusing it? Also test for empty role?

Reply:
The empty role would end up at a different http endpoint